app/components/pages/SubmissionWizard/steps/Files/SupportingUpload.js

```diff
@@ -185,11 +193,13 @@ class SupportingUpload extends React.Component {
     loop(iterator.next())
   }
 
-  updateFileState = (fileId, newState) => {
+  updateFileState = (fileId, newState, id) => {
     const newFilesState = [...this.state.files]
     const fileIndex = newFilesState.findIndex(file => file.id === fileId)
     if (fileIndex > -1) {
-      newFilesState[fileIndex] = { ...newFilesState[fileIndex], ...newState }
+      const updatingState = { ...newFilesState[fileIndex] }
+      updatingState.file.id = id
```

Comment:
why do we pass in `id` ? ... How is it different from `fileId`

Reply:
id is the id from the dataBase, we need to pass it to store id within the state, fileId is the array id and it's being used to remove loading spinner, so basically to know wich row is being updated.